one_fm/operations/doctype/operations_post/operations_post.py

```diff
@@ -60,49 +60,51 @@ def delete_schedule(doc):
     frappe.db.commit()
 
 def create_post_schedule_for_operations_post(operations_post):
-	contracts = get_active_contracts_for_project(operations_post.project)
-	if contracts:
-		if contracts.end_date >= getdate():
-			try:
-				owner = frappe.session.user
-				creation = now()
-				query = """
-					Insert Into
-						`tabPost Schedule`
-						(
-							`name`, `post`, `operations_role`, `post_abbrv`, `shift`, `site`, `project`, `date`, `post_status`,
-							`owner`, `modified_by`, `creation`, `modified`, `paid`, `unpaid`
-						)
-					Values
-				"""
-				post_abbrv = frappe.db.get_value("Operations Role", operations_post.post_template, ["post_abbrv"])
-				naming_series = NamingSeries('PS-')
-				ps_name_idx = previous_series = naming_series.get_current_value()
-				today = getdate()
-				start_date = today if contracts.start_date < today else contracts.start_date
-				for date in	pd.date_range(start=start_date, end=contracts.end_date):
-					if not frappe.db.exists("Post Schedule", {"date": cstr(date.date()), "post": operations_post.name}):
-						ps_name_idx += 1
-						ps_name = 'PS-'+str(ps_name_idx).zfill(5)
-						query += f"""
-							(
-								"{ps_name}", "{operations_post.name}", "{operations_post.post_template}", "{post_abbrv}",
-								"{operations_post.site_shift}", "{operations_post.site}", "{operations_post.project}",
-								'{cstr(date.date())}', 'Planned', "{owner}", "{owner}", "{creation}", "{creation}", '0', '0'
-							),"""
-				if previous_series == ps_name_idx:
-					frappe.msgprint(_("Post is already scheduled."))
-				else:
-					frappe.db.sql(query[:-1], values=[], as_dict=1)
-					frappe.db.commit()
-					naming_series.update_counter(ps_name_idx)
-					frappe.msgprint(_("Post is scheduled as Planned."))
-			except Exception as e:
-				frappe.db.rollback()
-				frappe.log_error('Post Schedule from Operations Post', e)
-				frappe.msgprint(_("Error log is added."), alert=True, indicator='orange')
-				operations_post.reload()
-		else:
-			frappe.msgprint(_("End date of the contract referenced in by the project is less than today."))
-	else:
-		frappe.msgprint(_("No active contract found for the project referenced."))
+    if type(operations_post) == str:
+        operations_post = frappe.get_doc("Operations Post",operations_post)
```

Comment:
Instead of this change in the Operations Post, set a method in contracts to send the operations post object. So that this file and method will not get changed and no related operations will be effected

Reply:
@pjamsheer Thank you for your contribution, I have modified the function as required, Kindly review.